File: src/main/vrml/net/coderextreme/x3d_code/www_web3d_org/x3d/content/examples/Basic/course/GroupFormation.js
let browser = X3D.getBrowser();
let X3D0 = {};
X3D0.profile = "Interchange";
X3D0.version = "3.0";
let WorldInfo2 = browser.currentScene.createNode("WorldInfo");
WorldInfo2.title = "GroupFormation.x3d";
browser.currentScene.children = new MFNode();

browser.currentScene.children[0] = WorldInfo2;

let Viewpoint3 = browser.currentScene.createNode("Viewpoint");
Viewpoint3.description = "watch the boxes";
Viewpoint3.position = new SFVec3f(new float[0,2,10]);
browser.currentScene.children[1] = Viewpoint3;

//Translate and rotate the entire formation here. 90 degrees around y axis turns to West.
let Transform4 = browser.currentScene.createNode("Transform");
Transform4.rotation = new SFRotation(new float[0,1,0,-0.345]);
let Group5 = browser.currentScene.createNode("Group");
Group5.DEF = "LEAD_ENTITY";
let Shape6 = browser.currentScene.createNode("Shape");
Shape6.DEF = "FLYING_BOX";
let Box7 = browser.currentScene.createNode("Box");
Box7.size = new SFVec3f(new float[0.5,0.5,3]);
Shape6.geometry = Box7;

let Appearance8 = browser.currentScene.createNode("Appearance");
let Material9 = browser.currentScene.createNode("Material");
Material9.diffuseColor = new SFColor(new float[0,0.2,0.8]);
Appearance8.material = Material9;

Shape6.appearance = Appearance8;

Group5.children = new MFNode();

Group5.children[0] = Shape6;

Transform4.children = new MFNode();

Transform4.children[0] = Group5;

let Transform10 = browser.currentScene.createNode("Transform");
Transform10.DEF = "SECOND_ENTITY";
Transform10.translation = new SFVec3f(new float[5,0,-2]);
let Shape11 = browser.currentScene.createNode("Shape");
Shape11.USE = "FLYING_BOX";
Transform10.children = new MFNode();

Transform10.children[0] = Shape11;

Transform4.children[1] = Transform10;

let Transform12 = browser.currentScene.createNode("Transform");
Transform12.DEF = "THIRD_ENTITY";
Transform12.translation = new SFVec3f(new float[-5,0,-2]);
let Shape13 = browser.currentScene.createNode("Shape");
Shape13.USE = "FLYING_BOX";
Transform12.children = new MFNode();

Transform12.children[0] = Shape13;

Transform4.children[2] = Transform12;

browser.currentScene.children[2] = Transform4;

File: src/main/python/net/x3djsonld/data/c/x3d-code/www.web3d.org/x3d/content/examples/Basic/course/GroupFormation.py
###############################################
#
# Now available: developmental python x3d.py package on PyPi for import.
#   This approach greatly simplifies Python X3D deployment and use.
#   https://pypi.org/project/x3d
#
# Installation:
#       pip install x3d
# or
#       python -m pip install x3d
#
# Developer options for loading x3d package:
#
#    from x3d import *  # preferred approach, terser source that avoids x3d.* class prefixes
#
# or
#    import x3d         # traditional way to subclass x3d package, all classes require x3d.* prefix
#                       # but python source is very verbose, for example x3d.Material x3d.Shape etc.
#                       # X3dToPython.xslt stylesheet insertPackagePrefix=true supports this option.

from x3d import *

###############################################

newModel=X3D(profile='Interchange',version='3.0',
  head=head(
    children=[
    meta(content='GroupFormation.x3d',name='title'),
    meta(content='Don Brutzman',name='creator'),
    meta(content='8 November 2001',name='created'),
    meta(content='20 October 2019',name='modified'),
    meta(content='Simple example to demonstrate rotating a Group of copied, elongated Boxes.',name='description'),
    meta(content='https://www.web3d.org/x3d/content/examples/Basic/course/GroupFormation.x3d',name='identifier'),
    meta(content='X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit',name='generator'),
    meta(content='../license.html',name='license')]),
  Scene=Scene(
    children=[
    WorldInfo(title='GroupFormation.x3d'),
    Viewpoint(description='watch the boxes',position=(0,2,10)),
    #  Translate and rotate the entire formation here. 90 degrees around y axis turns to West. 
    Transform(rotation=(0,1,0,-0.345),
      children=[
      Group(DEF='LEAD_ENTITY',
        children=[
        Shape(DEF='FLYING_BOX',
          geometry=Box(size=(.5,.5,3)),
          appearance=Appearance(
            material=Material(diffuseColor=(0,.2,.8))))]),
      Transform(DEF='SECOND_ENTITY',translation=(5,0,-2),
        children=[
        Shape(USE='FLYING_BOX')]),
      Transform(DEF='THIRD_ENTITY',translation=(-5,0,-2),
        children=[
        Shape(USE='FLYING_BOX')])])])
) # X3D model complete

###############################################
# Self-test diagnostics
###############################################

if        metaDiagnostics(newModel): # built-in utility method in X3D class
    print(metaDiagnostics(newModel))
print('check  newModel.XML() serialization...')
newModelXML = newModel.XML() # test export method XML() for exceptions
# print(newModelXML) # debug

print ("python x3d.py load successful for GroupFormation.py")
